pyth-rng/src/config/register_provider.rs

```diff
@@ -0,0 +1,24 @@
+use {
+    crate::config::{
+        EthereumOptions,
+        RandomnessOptions,
+    },
+    clap::Args,
+    ethers::types::U256,
+};
+
+#[derive(Args, Clone, Debug)]
+#[command(next_help_heading = "Register Provider Options")]
+#[group(id = "RegisterProvider")]
+pub struct RegisterProviderOptions {
+    #[command(flatten)]
```

Comment:
I'm surprised this works, I'm pretty sure they'll get argument name conflicts due to flatten, if this compiles though ignore this comment, but I have a feeling it might trip up the help command if you try and `--help`, you'll likely get a runtime error. There isn't a great way to work around this but one common pattern would be to lift the common arguments up to before the command when they're common to multiple methods.

Reply:
yeah i dunno. --help seems to work just fine. (I copied this from hermes)